Documents/Command/ConfigureAgentLogsUploadToCloudwatch/ConfigureAgentLogsUploadToCloudwatch.json

```diff
@@ -1,257 +1,326 @@
 {
-    "schemaVersion":"2.2",
-    "description":"Enable or Disable Pushing Agent Logs To CloudWatchLogs For Diagnostic Purposes",
-    "parameters": {
-      "LogLevel": {
-        "type": "String",
-        "description": "Log Level of SSM Agent",
-        "allowedValues": [
-          "Debug",
-          "Info",
-          "Error"
-        ]
+  "schemaVersion":"2.2",
+  "description":"Enable or Disable Pushing Agent Logs To CloudWatchLogs For Diagnostic Purposes",
+  "parameters": {
+    "LogLevel": {
+      "type": "String",
+      "description": "Log Level of SSM Agent",
+      "allowedValues": [
+        "Debug",
+        "Info",
+        "Error"
+      ]
+    },
+    "EnableCloudWatchUpload": {
+      "type": "String",
+      "description": "Enable Upload of Agent Logs To CloudWatch",
+      "allowedValues": [
+        "true",
+        "false"
+      ]
+    },
+    "LogGroup": {
+      "type": "String",
+      "default": "",
+      "description": "(Optional) Log Group Name. Will use the default group name if not specified. The log group would be created if not already present."
+    },
+    "EnableShare": {
+      "type": "String",
+      "default": "false",
+      "description": "(Optional) Enable sharing of logs with CloudWatch of another AWS Account. The ShareDestination parameter needs to be passed if true.",
+      "allowedValues": [
+        "true",
+        "false"
+      ]
+    },
+    "ShareDestination": {
+      "type": "String",
+      "default": "",
+      "description": "(Optional) Destination in the format 'IAMUserAccessKeyID::IAMUserSecretAccessKey::DestinationLogGroup::DestinationLogStream'. The IAM User should have access to DescribeLogStreams, PutLogEvents on the destination CloudWatchLogs resource. The log group and stream should be present. Parameter will be ignored if EnableShare is false",
+      "allowedPattern": "(^[\\w]+::.+::[\\.\\-_/#A-Za-z0-9]+::[^:*]+$)|^$"
+    }
+  },
+  "mainSteps": [
+    {
+      "action":"aws:runPowerShellScript",
+      "precondition": {
+        "StringEquals": ["platformType", "Windows"]
       },
-      "EnableCloudWatchUpload": {
-        "type": "String",
-        "description": "Enable Upload of Agent Logs To CloudWatch",
-        "allowedValues": [
-          "true",
-          "false"
+      "name":"UploadSSMAgentLogsInWindows",
+      "inputs": {
+        "runCommand":[
+          "$loglevel = '{{ LogLevel }}'",
+          "[bool] $enable = ${{ EnableCloudWatchUpload }}",
+          "$logGroup = '{{ LogGroup }}'",
+          "[bool] $enableSharing = ${{ EnableShare }}",
```

Comment:
Do input validation, do not directly use the parameter {{ EnableShare }} as a variable name.
Example:

`
$EnableShareInput = "{{ EnableShare }}"
switch ($EnableShareInput) {
        "true" { $enableSharing = $True }
        "false" { $enableSharing = $False }
        default { throw "Unexpected input."} 
}
`

Reply:
Yes, having validation sounds good. I'll fix that.